src/main/java/crazypants/enderio/teleport/TravelController.java

```diff
@@ -190,8 +193,27 @@ public void addBlockToBlinkBlackList(String blockName) {
     }
 
     public boolean activateTravelAccessable(ItemStack equipped, World world, EntityPlayer player, TravelSource source) {
+        return activateTravelAccessable(equipped, world, player, source, false);
+    }
+
+    public boolean activateTravelAccessable(ItemStack equipped, World world, EntityPlayer player, TravelSource source,
+            boolean alsoDoTeleport) {
         if (!hasTarget()) {
             PacketLongDistanceTravelEvent p = new PacketLongDistanceTravelEvent(player, false, source);
+            if (alsoDoTeleport) {
+                Optional<BlockCoord> destinationOptional = findTeleportDestination(player);
+                if (destinationOptional.isPresent()) {
+                    BlockCoord destination = destinationOptional.get();
+                    p = new PacketLongDistanceTravelEvent(
+                            player,
+                            false,
+                            source,
+                            true,
+                            destination.x,
+                            destination.y,
+                            destination.z);
+                }
+            }
```

Comment:
I think that doesn't quite work, because we actually have two nested `if`-statements (lines 203 and 205). We'd have to either have duplicate `else`-clauses, or always call the expensive `findTeleportDestination()` method even if it's not needed.

I think the extra allocation is probably cheap enough not to warrant the duplicate `else`-clauses, so I've left this as-is. We could also avoid the extra allocation by using `Supplier<PacketLongDistanceTravelEvent>` instead, though I don't know if that would actually end up even more expensive due to allocating a `Supplier` and adding a call to `Supplier.get()`.

Reply:
```java
            PacketLongDistanceTravelEvent p;
            if (alsoDoTeleport) {
                Optional<BlockCoord> destinationOptional = findTeleportDestination(player);
                if (destinationOptional.isPresent()) {
                    BlockCoord destination = destinationOptional.get();
                    p = new PacketLongDistanceTravelEvent(
                            player,
                            false,
                            source,
                            true,
                            destination.x,
                            destination.y,
                            destination.z);
                }
            }
            if (p == null) {
                // Forward teleport disabled or impossible, only teleport to an anchor.
                p = new PacketLongDistanceTravelEvent(player, false, source);
            }
```

Reply:
Sorry, I missed your most recent comment. How's this?

Reply:
Oh, that's even simpler 👍